### PR TITLE
Map nullables

### LINKED ISSUE #79: Add `Map` and `Bind` extensions to nullable structs, and `Choose` to `IEnumerable<nullable>` collections.

### PR DESCRIPTION
**Pull request by JYCabello**
closes #79

### CODE REVIEW

File: src/DeFuncto.Tests/Core/Extensions/Nullables/Bind.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    {
+        var result = value.Bind(v => anotherValue.Map(av => av + v));
+
+        if (value is not null && anotherValue is not null)
+        {
+            Assert.NotNull(result);
+            Assert.Equal(value.Value + anotherValue.Value, result.Value);
+        }
+
+        if (value is null || anotherValue is null)
```
Comment (theonlyroboboro) [inline src/DeFuncto.Tests/Core/Extensions/Nullables/Bind.cs:20]: else if is faster

**Review by theonlyroboboro — APPROVED** [review]

**Comment by theonlyroboboro** [comment]
SonarCloud is not working?

File: src/DeFuncto.Tests/Core/Extensions/Nullables/Bind.cs (hunk 1 above)
Comment (JYCabello) [reply src/DeFuncto.Tests/Core/Extensions/Nullables/Bind.cs:20]: Care to explain?

Comment (JYCabello) [reply src/DeFuncto.Tests/Core/Extensions/Nullables/Bind.cs:20]: Discussed in whatsapp, merging.

**Review by theonlyroboboro — APPROVED** [review]

**Pull request merged**